packages/remirror__extension-list/src/checkbox-extension.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  ApplySchemaAttributes,
+  command,
+  CommandFunction,
+  extension,
+  ExtensionTag,
+  getMatchString,
+  InputRule,
+  isElementDomNode,
+  isNodeSelection,
+  NodeExtension,
+  NodeExtensionSpec,
+  nodeInputRule,
+  NodeSpecOverride,
+  omitExtraAttributes,
+} from '@remirror/core';
+import type { CreateEventHandlers } from '@remirror/extension-events';
+import { NodeSelection } from '@remirror/pm/state';
+
+/**
+ * Creates the node for a list item.
+ */
+@extension({})
+export class CheckboxExtension extends NodeExtension {
+  get name() {
+    return 'checkbox' as const;
```

Comment:
This is it's own extension, aiming to achieve the requirement of allow custom rendering of checkbox items.